src/structure_managers/property_block_sparse.hh

```diff
@@ -0,0 +1,600 @@
+/**
+ * file   property_base.hh
+ *
+ * @author Till Junge <till.junge@epfl.ch>
+ * @author Felix Musil <felix.musil@epfl.ch>
+ *
+ * @date   03 Aug 2018
+ *
+ * @brief implementation of non-templated base class for Properties, Properties
+ *        are atom-, pair-, triplet-, etc-related values
+ *
+ * Copyright © 2018 Till Junge, Felix Musil, COSMO (EPFL), LAMMM (EPFL)
+ *
+ * Rascal is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation, either version 3, or (at
+ * your option) any later version.
+ *
+ * Rascal is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this software; see the file LICENSE. If not, write to the
+ * Free Software Foundation, Inc., 59 Temple Place - Suite 330,
+ * Boston, MA 02111-1307, USA.
+ */
+
+#ifndef SRC_STRUCTURE_MANAGERS_PROPERTY_BLOCK_SPARSE_HH_
+#define SRC_STRUCTURE_MANAGERS_PROPERTY_BLOCK_SPARSE_HH_
+
+#include "structure_managers/property_base.hh"
+#include "structure_managers/cluster_ref_key.hh"
+
+#include <unordered_map>
+#include <set>
+#include <list>
+#include <map>
+#include <algorithm>
+#include <iterator>
+
+namespace rascal {
+
+  namespace internal {
+
+    template <class K, class V>
+    class InternallySortedKeyMap {
+     public:
+      using MyMap_t = std::map<K, V>;
+
+      //! the data holder. only the overiden/essential functionalities are
+      //! directly exposed
+      MyMap_t data{};
+
+      // some member types
```

Comment:
Member types in camelcase or is this for compliance with std library?

Reply:
yes